lib/statsd.rb

```diff
@@ -125,9 +137,22 @@ def sampled(sample_rate)
 
   def send(stat, delta, type, sample_rate=1)
     sampled(sample_rate) do
-      prefix = "#{@namespace}." unless @namespace.nil?
-      stat = stat.to_s.gsub('::', '.').gsub(RESERVED_CHARS_REGEX, '_')
-      msg = "#{prefix}#{stat}:#{delta}|#{type}#{'|@' << sample_rate.to_s if sample_rate < 1}"
+      stat = stat.to_s.dup
+      stat.gsub!(COLON_COLON_REGEX, ".".freeze)
```

Comment:
Hm, my reading is that it does indeed allocate a Regexp:

https://github.com/github/ruby/blob/b3255a8b572160bf249b5019c1ec93e667d547ee/string.c#L4171
https://github.com/github/ruby/blob/b3255a8b572160bf249b5019c1ec93e667d547ee/string.c#L3976-L3978